src/org/labkey/targetedms/folderImport/PanoramaQCSettings.java

```diff
@@ -53,8 +54,10 @@ public void exportSettings(VirtualFile vf, Container container, User user) throw
                     List<ColumnInfo> userEditableCols = ti.getColumns().stream().filter(ci -> ci.isUserEditable()).collect(Collectors.toList());
                     SimpleFilter filter = SimpleFilter.createContainerFilter(container); //only export the ones that are defined in current container (and not the ones from the root container)
 
-                    ResultsFactory factory = ()-> QueryService.get().select(ti, userEditableCols, filter, null);
+                    Results results = QueryService.get().select(ti, userEditableCols, filter, null);
+                    ResultsFactory factory = ()-> results;
                     exportSettingsToTSV(vf, factory, getSettingsFileName(), getTableName());
+                    results.close();
```

Comment:
This and the six other similar changes in this commit don't look right to me. The whole point of passing a factory into `exportSettingsToTSV()` (which then passes it into `TSVGridWriter`) is to allow the calling code to determine how a `Results` is generated, but to give the responsibility for actually generating and closing the `Results` to the consuming code (TSVGridWriter, in this case). Passing in a factory allows the consumer to use try-with-resources to handle the results (much preferred over passing around open Results objects). This change is not going to fix any "CachedResultSet not closed" errors... invoking the factory and closing that Results here just executes an extra query and throws away the results; it doesn't in any way affect the Results generated by TSVGridWriter by another invocation of the factory.

Now I see the problem... `exportSettingsToTSV()` inovkes `factory.get().countAll()`... this call to the factory generates the CachedResultSet that's never closed. I'm sure you want to execute the query only once (current code is executing it three times), so generate and stash a Results, check countAll(), and pass a new factory `()->results` to TSVGridWriter. You have to close the results in the count == 0 case, so might as well generate the Results in a try-with-resources. This means it will be double closed in most cases, but CachedResultSet doesn't care.

Reply:
This is difficult to explain in a commit comment... happy to discuss, just let me know

Reply:
This change did fix the "CachedResultSet not closed" errors: 
https://teamcity.labkey.org/viewType.html?buildTypeId=LabKey_223Release_Premium_ModulesSuites_PanoramaPremiumPostgres&branch_LabKey_223Release_Premium_ModulesSuites=copy_qc_folders&tab=buildTypeStatusDiv

Sounds good, lets sync on Monday! Thanks!

Reply:
We discussed today, but for the record: I missed that invoking the factory doesn't execute the query... so it makes sense that this change fixed the exception and doesn't actually execute the query multiple times (as I had claimed). I still think it would be cleaner for each caller to simply pass in a query-executing factory and for `exportSettingsToTSV()` to invoke the factory in a try-with-resources, check countAll(), and pass on a `()->results` factory if appropriate.

Reply:
Hi Adam - just seeing your followup comment, thanks for clarifying and insights. I've made some updates, please review. Thank you!